src/js/controllers/modals/inputAmount.js

```diff
@@ -0,0 +1,180 @@
+        return val + operator;
+      } else {
+        return val.slice(0, -1) + operator;
+      }
+    };
+  };
+
+  function isOperator(val) {
+    var regex = /[\/\-\+\x\*]/;
+    var match = regex.exec(val);
+    if (match) return true;
+    return false;
+  };
+
+  function isExpression(val) {
+    var regex = /^\.?\d+(\.?\d+)?([\/\-\+\*x]\d?\.?\d+)+$/;
+    var match = regex.exec(val);
+
+    if (match) return true;
+    return false;
+  };
+
+  $scope.removeDigit = function() {
+    if ($scope.amount.toString().length == 1) {
+      $scope.resetAmount();
+      return;
+    }
+
+    $scope.amount = $scope.amount.slice(0, -1);
+    processAmount($scope.amount);
+    checkFontSize();
+  };
+
+  $scope.resetAmount = function() {
+    $scope.amount = $scope.alternativeResult = $scope.amountResult = $scope.globalResult = '';
+    checkFontSize();
+  };
+
+  function processAmount(val) {
+    if (!val) {
+      $scope.resetAmount();
+      return;
+    }
+
+    var formatedValue = format(val);
+    var result = evaluate(formatedValue);
+
+    if (lodash.isNumber(result)) {
+      if ($scope.showAlternativeAmount)
+        $scope.globalResult = isExpression(val) ? '= ' + $filter('formatFiatAmount')(result) : '';
+      else
+        $scope.globalResult = isExpression(val) ? '= ' + profileService.formatAmount(result * unitToSatoshi, true) : '';
+
+      $scope.amountResult = $filter('formatFiatAmount')(toFiat(result));
+      $scope.alternativeResult = profileService.formatAmount(fromFiat(result) * unitToSatoshi, true);
+    }
+  };
+
+  function fromFiat(val) {
+    return parseFloat((rateService.fromFiat(val, $scope.alternativeIsoCode) * satToUnit).toFixed(unitDecimals), 10);
+  };
+
+  function toFiat(val) {
+    return parseFloat((rateService.toFiat(val * unitToSatoshi, $scope.alternativeIsoCode)).toFixed(2), 10);
+  };
+
+  function evaluate(val) {
+    var result;
+    try {
+      result = $scope.$eval(val);
+    } catch (e) {
+      return 0;
+    }
+    if (result == 'Infinity' || lodash.isNaN(result)) return 0;
+    return result;
+  };
+
+  function format(val) {
+    var result = val.toString();
+
+    if (isOperator(lodash.last(val)))
+      result = result.slice(0, -1);
+
+    return result.replace('x', '*');
+  };
+
+  $scope.finish = function() {
+    var amount;
+    var alternativeAmount;
+
+    if ($scope.showAlternativeAmount) {
+      amount = fromFiat($scope.globalResult);
+      alternativeAmount = profileService.formatAmount(evaluate(format($scope.amount)) * unitToSatoshi, true);
+    } else {
+      amount = profileService.formatAmount(evaluate(format($scope.amount)) * unitToSatoshi, true);
+      alternativeAmount = toFiat(amount);
+    }
+
+    if ($scope.address) {
+      var satToBtc = 1 / 100000000;
+      var amountSat = parseInt((amount * unitToSatoshi).toFixed(0));
+      if ($scope.unitName == 'bits') {
+        $scope.specificAmountBtc = (amountSat * satToBtc).toFixed(8);
```

Comment:
Why is this done when `unitName` == 'bits'? Also, it seems `specificAmountBtc` is not used. Is this dead code?